3.PHPBasics/app/views/index.php

```diff
@@ -0,0 +1,199 @@
+                    <ol>
+                        <?php foreach ($text->getShortestSentences() as $sentence) : ?>
+                            <li><?= $sentence ?></li>
+                        <?php endforeach; ?>
+                    </ol>
+                </section>
+            </div>
+            <!-- -->
+            <section>
+                <div class="row">
+                    <div class="col-6 col-12-medium" style="padding-right: 10px">
+                        <h2>Frequency of characters</h2>
+                        <div class="table-wrapper">
+                            <table>
+                                <thead>
+                                <tr>
+                                    <th>Character</th>
+                                    <th>Frequency</th>
+                                </tr>
+                                </thead>
+                                <tbody>
+                                <?php foreach ($text->getCharactersFrequency() as $char => $count) : ?>
+                                    <tr>
+                                        <td><?= $char ?></td>
+                                        <td><?= $count ?></td>
+                                    </tr>
+                                <?php endforeach; ?>
+                                </tbody>
+                            </table>
+                        </div>
+                    </div>
+                    <div class="col-6 col-12-medium">
+                        <h2>Distribution of characters as a percentage of total</h2>
+                        <div class="table-wrapper">
+                            <table>
+                                <thead>
+                                <tr>
+                                    <th>Character</th>
+                                    <th>Percentage</th>
+                                </tr>
+                                </thead>
+                                <tbody>
+                                <?php foreach ($text->getCharactersPercentage() as $char => $percentage) : ?>
+                                    <tr>
+                                        <td><?= $char ?></td>
+                                        <td><?= round($percentage, 2) . ' %' ?></td>
+                                    </tr>
+                                <?php endforeach; ?>
+                                </tbody>
+                            </table>
+                        </div>
+                    </div>
+                </div>
+            </section>
+            <h3>The reversed text:</h3>
+            <blockquote><?= $text->getReverseText() ?></blockquote>
+            <!-- -->
+            <h3>The reversed text but the character order in words kept intact:</h3>
+            <blockquote><?= $text->getReverseWords() ?></blockquote>
+            <p>Date and time when the report was generated: <?= isset($timeService) ? $timeService->getCurrentFormattedDataTime() : ''?> </p>
```

Comment:
that is why you make here check for `timeService isset` because it could be not initialized in controller, looks like this is mandatory service and we cannot work without correctly :)